statsmodels/discrete/tests/test_discrete.py

```diff
@@ -100,13 +107,39 @@ def test_bic(self):
         assert_almost_equal(self.res1.bic, self.res2.bic, DECIMAL_3)
 
     def test_predict(self):
-        assert_almost_equal(self.res1.model.predict(self.res1.params),
-                            self.res2.phat, DECIMAL_4)
+
+        fitted = self.res1.model.predict(self.res1.params)
+
+        res2_fitted = getattr(self.res2, 'phat', None)
+        if res2_fitted is None:
+            res2_fitted = np.exp(getattr(self.res2, 'linpred', None))
+        if res2_fitted is not None:
+            assert_allclose(fitted[:len(res2_fitted)], res2_fitted, rtol=5e-4)
+
+        # fittedvalues in discrete are currently linear prediction
+        assert_allclose(self.res1.fittedvalues[:len(res2_fitted)],
+                        res2_fitted, atol=5e-4, rtol=5e-4)
+
+        if self.res1.params.ndim == 2:
+            # special case for MNLogit
+            endog = self.res1.model.wendog
+        else:
+            endog = self.res1.model.endog
+        resid = (endog - fitted)
+        assert_allclose(self.res1.resid, resid, atol=1e-10)
+        assert_allclose(self.res1.resid, self.res1.resid_response, atol=1e-10)
 
     def test_predict_xb(self):
-        assert_almost_equal(self.res1.model.predict(self.res1.params,
-                            linear=True),
-                            self.res2.yhat, DECIMAL_4)
+        try:
+            linpred = self.res1.predict(linear=True)
+        except TypeError:
+            # new count models do not have 'linear' as keyword
+            linpred = self.res1.predict(which='linear')
+
+        res2_linpred = getattr(self.res2, 'linpred', None)
+        if res2_linpred is not None:
+            assert_allclose(linpred[:len(res2_linpred)], res2_linpred,
+                            atol=5e-4, rtol=1e-4)
```

Comment:
ATM `TestNegativeBinomialPNB2Newton.test_predict_xb` has `rtol=-1e-7`.  In the PR^2 I'll offer an alternative to keep this tighter tolerance (... after I get back from lunch)